src/proxy-ui-api/src/main/java/org/niis/xroad/restapi/service/BackupService.java

```diff
@@ -147,6 +155,44 @@ public BackupFile generateBackup() throws InterruptedException {
         return backupFile.get();
     }
 
+    /**
+     * Write uploaded backup file to disk. If ignoreWarnings=false, an exception is thrown when a file with
+     * the same name already exists. If ignoreWarnings=true, the existing file is overwritten.
+     * @param ignoreWarnings
+     * @param file
+     * @return
+     * @throws InvalidFilenameException if backup file's name is invalid and does not pass validation
+     * @throws UnhandledWarningsException if backup file with the same name already exists
+     * and ignoreWarnings is false
+     * @throws InvalidBackupFileException if backup file is not a valid tar file or the first entry of the tar file
+     * does not match to the first entry if the Security Server generated backup tar files
+     */
+    public BackupFile uploadBackup(Boolean ignoreWarnings, MultipartFile file)
```

Comment:
This method could take `String fileName` and `byte[] fileBytes` as parameters instead of a `MultipartFile`. I think that would create better separation between the controller and service layer and make this method more universal. What do you think?

Reply:
Passing `String fileName` and `byte[] fileBytes` as parameters instead of `MultipartFile` is a good idea. Updated the implementation.